python/lsst/ts/mtreflector/controller.py

```diff
@@ -0,0 +1,220 @@
+    """Implement MTReflector controller.
+
+    Parameters
+    ----------
+    config : `types.SimpleNamespace`
+        The configuration object.
+    log : `logging.Logger`
+        The log object.
+    simulation_mode : `int`
+        Controller in simulation mode?
+    Attributes
+    ----------
+    handle : `None` or `int`
+        The labjack ljm handle.
+    state : `lsst.ts.xml.MTReflector.MTReflectorStatus`
+        The current state of the reflector.
+    device_type : `int`
+        The labjack device type.
+    connection_type : `int`
+        The connection type to the labjack.
+    identifier : `str`
+        The host name of the labjack.
+    simulation_mode : `int`
+        Controller in simulation mode.
+    log : `logging.Logger`
+        The log object.
+    config : `types.SimpleNamespace`
+        The config object.
+    fake_value : `None` or `int`
+        The fake value used by simulation mode.
+    """
+
+    def __init__(
+        self,
+        config: types.SimpleNamespace,
+        log: logging.Logger | None,
+        simulation_mode: int,
+    ) -> None:
+        self.handle: None | int = None
+        self.state: MTReflectorStatus = MTReflectorStatus.UNKNOWN
+        self.device_type: int = ljm.constants.dtT4
+        self.connection_type: int = ljm.constants.ctTCP
+        self.identifier: None | str = None
+        self.simulation_mode: int = simulation_mode
+        self.log: logging.Logger = (
+            log if log is not None else logging.getLogger(__name__)
+        )
+        self.config: types.SimpleNamespace = config
+        self.fake_value: None | float = None
+        self._loop: asyncio.AbstractEventLoop = asyncio.get_event_loop()
+
+    async def run(self, func: Callable, **kwargs: Any) -> Any:
+        """Run the command.
+
+        Paramters
+        ---------
+        func : `Callable`
+            The function to run.
+        kwargs : `Any`
+            Any keyword arguments accepted by the function.
+
+        Raises
+        ------
+        If the function is incorrect.
+        If the execution fails.
+
+        Returns
+        -------
+        `Any`
+            Value returned if any.
+
+        """
+        try:
+            partial = functools.partial(func, **kwargs)
+        except Exception:
+            raise
+        try:
+            return await self._loop.run_in_executor(None, partial)
+        except Exception:
+            self.log.exception("Command failed.")
+            return None
+
+    @property
+    def connected(self) -> bool:
+        """Connected to labjack?"""
+        return self.handle is not None
+
+    def configure(self) -> None:
+        """Configure the controller."""
+        if not self.simulation_mode:
+            self.identifier = self.config.identifier
+        else:
+            self.identifier = ljm.constants.DEMO_MODE
+
+    async def connect(self) -> None:
+        """Connect to the labjack."""
+        self.configure()
+        try:
```

Comment:
If you move the `self.log.exception` line to the `write_channel` method, you can remove the try/exception block.

Reply:
If you
- remove the try/except block on lines 112 to 115
- raise RuntimeError in (the current) line 120
then you can get rid of the try/except in the `connect` method.